server/src/test/java/org/elasticsearch/search/SearchHitTests.java

```diff
@@ -353,6 +353,42 @@ public void testWeirdScriptFields() throws Exception {
         }
     }
 
+    public void testToXContentEmptyFields() throws IOException {
+        Map<String, DocumentField> fields = new HashMap<>();
+        fields.put("foo", new DocumentField("foo", Collections.emptyList()));
+        fields.put("bar", new DocumentField("bar", Collections.emptyList()));
+        SearchHit hit = new SearchHit(0, "_id", fields, Collections.emptyMap());
+        {
+            BytesReference originalBytes = toShuffledXContent(hit, XContentType.JSON, ToXContent.EMPTY_PARAMS, randomBoolean());
+            final SearchHit parsed;
+            try (XContentParser parser = createParser(XContentType.JSON.xContent(), originalBytes)) {
+                parser.nextToken(); // jump to first START_OBJECT
+                parsed = SearchHit.fromXContent(parser);
+                assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+                assertNull(parser.nextToken());
+            }
+            assertThat(parsed.getFields().size(), equalTo(0));
```

Comment:
Can we also check that the `fields` object is not only empty in the xContent rendering but really not present? I think the parsed object would contain an empty map regardless.

Reply:
I am not sure I understand, we check if the empty fields are removed when rendering to xcontent. What kind of test are you expecting ?

Reply:
Sorry that I wasn't clear, what I mean is that both hits:
```
{ 
    [...], 
    "_source" : { ... },
    "fields" : { }
}
```
and 
```
{ 
    [...], 
    "_source" : { ... }
}
```
would be parsed to a `SearchHit` where `getFields()` returns an empty map, but it would be nice to check that we really don't render the `fields` object at all when there are no entries to the documentFields with size>0. It's only about checking the output of the rendering step directly rather than parsing it, because then we cannot tell the difference in the test. Not sure if that's clearer now and if such a test is necessary but I think it would be good to add, but not a bit deal if not...

Reply:
Got it, thanks. I pushed https://github.com/elastic/elasticsearch/pull/58418/commits/459cb604e4464f8ee2978958b7ebaaff993dd7de